Assets/Scripts/CartesianTiles/WFSGeoJSONTileDataLayer.cs

```diff
@@ -0,0 +1,148 @@
+            }
+
+            StartCoroutine(FindTileHandler());
+        }
+
+        private IEnumerator FindTileHandler()
+        {
+            yield return null;
+
+            //Find a required TileHandler in our parent, or else in the scene
+            tileHandler = GetComponentInParent<TileHandler>();
+
+            if (!tileHandler)
+                tileHandler = FindAnyObjectByType<TileHandler>();
+
+            if (tileHandler)
+            {
+                tileHandler.AddLayer(this);
+                yield break;
+            }
+
+            Debug.LogError("No TileHandler found.", gameObject);
+        }
+
+        public override void HandleTile(TileChange tileChange, System.Action<TileChange> callback = null)
+        {
+            TileAction action = tileChange.action;
+            var tileKey = new Vector2Int(tileChange.X, tileChange.Y);
+            switch (action)
+            {
+                case TileAction.Create:
+                    Tile newTile = CreateNewTile(tileKey);
+                    tiles.Add(tileKey, newTile);
+                    newTile.runningCoroutine = StartCoroutine(DownloadGeoJSON(tileChange, newTile, callback));
+                    break;
+                case TileAction.Upgrade:
+                    tiles[tileKey].unityLOD++;
+                    break;
+                case TileAction.Downgrade:
+                    tiles[tileKey].unityLOD--;
+                    break;
+                case TileAction.Remove:
+                    geoJSONLayer.RemoveFeaturesOutOfView();
+
+                    InteruptRunningProcesses(tileKey);
+                    tiles.Remove(tileKey);
+                    callback?.Invoke(tileChange);
+                    return;
+                default:
+                    break;
+            }
+        }
+
+        private void OnGeoJSONLayerDestroyed()
+        {
+            Destroy(gameObject);
+        }
+
+        private void OnDestroy()
+        {
+            if (tileHandler)
+                tileHandler.RemoveLayer(this);
+        }
+
+        private Tile CreateNewTile(Vector2Int tileKey)
+        {
+            Tile tile = new()
+            {
+                unityLOD = 0,
+                tileKey = tileKey,
+                layer = transform.gameObject.GetComponent<Layer>()
+            };
+
+            return tile;
+        }
+
+        private IEnumerator DownloadGeoJSON(TileChange tileChange, Tile tile, System.Action<TileChange> callback = null)
+        {
+            var bboxValue = $"{tileChange.X},{tileChange.Y},{(tileChange.X + tileSize)},{(tileChange.Y + tileSize)}";
+            string url = WfsUrl.Replace("{0}", bboxValue);
+            Debug.Log("Downloading GeoJSON from: " + url);
+
+            var geoJsonRequest = UnityWebRequest.Get(url);
+            tile.runningWebRequest = geoJsonRequest;
+            yield return geoJsonRequest.SendWebRequest();
+
+            if (geoJsonRequest.result == UnityWebRequest.Result.Success)
```

Comment:
Er zit geen error handling in dit stuk; is dat iets wat we laten en accepteren dat er dan geen tegel getoond wordt? Of moeten we bij Friso en Sterre ook een vraagstuk neerleggen hoe we hiermee om zouden kunnen gaan? Ik denk dan aan out of bounds issues, of zelfs simpelweg een timeout omdat iemand dit op zijn laptop in de trein doet en een tunnel ingaat.

Reply:
Word los kaartje